### ISSUE TITLE
Chart markers

### ISSUE THREAD

**Issue by itsderek23** [scoutapp/roadmap#4]
Add the ability to add markers to charts to identify things like the begin/end of outages.

**Comment by lmansur** [comment]
+1

Our deploy process takes a while to complete so it would be great if we could add a custom marker (via API preferably) as soon as the deploy starts and measure how much it impacts our response time.

**Comment by itsderek23** [comment]
Just to clarify: this is in addition to our [deploy markers](http://help.apm.scoutapp.com/#deploy-tracking-config) @lmansur? These markers would appear after a newly deployed revision is deployed to any app server.

**Comment by lmansur** [comment]
Yes @itsderek23. We would like to add a Custom Marker as soon as we start deploying instead of as soon as a new app server is started.

**Comment by pokrovskyy** [comment]
+1, suggested the same thing on Scout Slack a minute ago. This would allow making a custom note on the chart eg. upgraded / reconfigured server hardware / software etc. or some other external event not necessarily recorded in git. And then see how it affected the performance over time

**Comment by ideasasylum** [comment]
Just came across this issue because I was looking for a way to annotate charts with configuration changes (in this case, scaling dynos, changing WEB_CONCURRENCY & max number of Puma threads etc). Although Heroku deploys a new version when the environment variables change, this isn't reflected in Scout because it uses the Git SHA (AFAIK).

Wouldn't be nice just to note these changes on the chart
